pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -3304,10 +3305,19 @@ private CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final TopicName
                     topicName.getNamespaceObject());
             return CompletableFuture.completedFuture(false);
         }
-        //System topic can always be created automatically
+
+        // ServiceUnitStateChannelImpl.TOPIC expects to be a non-partitioned-topic now.
+        // We don't allow the auto-creation here.
+        // ServiceUnitStateChannelImpl.start() is responsible to create the topic.
+        if (ServiceUnitStateChannelImpl.TOPIC.equals(topicName.toString())) {
```

Comment:
This change makes sense to me, but we should finally change it to a partitioned topic.

Reply:
Yes this is another optimization area for scalability.

Reply:
Why is it that in some cases we auto create in a non-partitioned way and in this case we do not auto create it?

Reply:
I am not sure if I understood your question. Since the system topic is currently expected to work on a non-partitioned topic, we specifically create the topic in ServiceUnitStateChannelImpl.start(). This change will block the system topic auto-creation(when any other broker tries to auto-create the topic due to any race-condition )

Reply:
Sorry, I was mixing up https://github.com/apache/pulsar/pull/20370 and https://github.com/apache/pulsar/pull/20397.

My primary concern is that the solution in this PR creates a special case instead of creating a framework to make it easier to build system topics in the future. In my opinion, the load manager should be contained by the relevant interfaces and should not be referenced here. There is some relevant discussion about system topics in this PR too https://github.com/apache/pulsar/pull/20514#issuecomment-1579937937. There is also relevant discussion on the ML here https://lists.apache.org/thread/f0q8n0hf1lgw9r2j53tm4yjjfdyr9kjd.

I think it is relevant that until now, we have always allowed system topics to be auto created to prevent classes of  failures.

If I were to guess, the problem this PR is trying to solve is actually more generic than the system topics we're working with here. It is likely a challenge for all pulsar users to know how to create their topics correctly. That is one reason I want to push for thinking about a general solution.

Reply:
I think we can make this fix general like the followings.

1. Maintain a White-list for system topics that are pre-created. We can check this list and reject any auto creation here.

2. Or Make Pulsar pre-create all system topics, and then we can reject auto-creation for all topics under the system namespace.

I think the first option is less intrusive.

Reply:
A third option is to only auto create topics as non-partitioned topics. Any component that needs to create a system topic as a partitioned topic could do so on start up.

Reply:
I proposed as much here on the ML: https://lists.apache.org/thread/1lndgf1hx821fc12t0pc6j6zdrhkntht.